modules/architecture-machine-config-pools.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * architecture/control-plane.adoc
+
+[id="architecture-machine-config-pools_{context}"]
+= Node configuration management with machine config pools
+
+Machines that manage the cluster and its workloads are divided into groups based on the types of resources they handle. These groups of machines are called machine config pools (MCP). Each MCP manages a set of nodes and its corresponding machine configs. The role of the node determines which MCP it belongs to; the MCP governs nodes based on its assigned node role label. Nodes in an MCP have the same configuration; this means nodes can be scaled up and torn down in response to increased or decreased workloads.
+
+By default, there are two MCPs created by the cluster when it is installed: `master` and `worker`. Each default MCP has a defined configuration applied by the Machine Config Operator (MCO), which is responsible for managing MCPs and facilitating MCP upgrades. You can create additional MCPs, or custom pools, to manage nodes that have custom use cases that extend outside of the default node types.
+
+[NOTE]
+====
+A node can only be included in one machine config pool.
+====
+
+Custom pools are pools that inherit their configurations from the worker pool. They use any machine config targeted for the worker pool, but add the ability to deploy changes only targeted at the custom pool. Since a custom pool inherits its configuration from the worker pool, any change to the worker pool is applied to the custom pool as well. Custom pools inheriting configurations from master nodes are not supported by the MCO.
```

Comment:
Maybe to add a bit more:
`Custom pools inheriting configurations from master nodes are not supported by the MCO` -> `Custom pools inheriting configurations from master nodes, or custom pools that do not inherit from any pool, are not supported by the MCO` or just `Custom pools not inheriting from worker pool are not supported by the MCO`